bullet_train-api/app/helpers/api/open_api_helper.rb

```diff
@@ -100,31 +100,54 @@ def automatic_components_for(model, **options)
       end
 
       if has_strong_parameters?("Api::#{@version.upcase}::#{model.name.pluralize}Controller".constantize)
-        strong_params_module = "Api::#{@version.upcase}::#{model.name.pluralize}Controller::StrongParameters".constantize
-        strong_parameter_keys = BulletTrain::Api::StrongParametersReporter.new(model, strong_params_module).report
-        if strong_parameter_keys.last.is_a?(Hash)
-          strong_parameter_keys += strong_parameter_keys.pop.keys
+        strong_params_module_name = "Api::#{@version.upcase}::#{model.name.pluralize}Controller::StrongParameters".constantize
+        strong_params_module = BulletTrain::Api::StrongParametersReporter.new(model, strong_params_module_name)
```

Comment:
So I went looking to see if these two variables are used anywhere else, just to make sure that it's safe change their semantic meaning like this, and I came across this other block of code that is _very_ similar to the original code in this block.

https://github.com/bullet-train-co/bullet_train-core/blob/567d4ad9b8e7a9314e278f6a17bc26399630c3fb/bullet_train-api/lib/bullet_train/api/example_bot.rb#L45-L61

Do we need to also update that one to match this one?

If not we may want to at least reconsider changing the semantic meaning of these variables in this block. I think it would be nice for both blocks to have variables that mean more or less the same thing so that there's less context switching required to work on them.

Reply:
Here we should probably get different `strong_parameter_keys` based on method as well. Feels like this functionality should be extracted to a separate method to avoid code duplicating.